src/TMS/dev/TMP117.cpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by Ishan on 1/23/2023.
+//
+
+#include <TMS/dev/TMP117.hpp>
+#include <EVT/io/I2C.hpp>
+
+namespace TMS {
+
+TMP117::TMP117(IO::I2C& i2c, uint8_t deviceID): deviceID(deviceID), i2c(i2c) { }
+
+uint16_t TMP117::readTemp() {
+    uint16_t input;
+    uint8_t tempBytes[2];
+    uint16_t byte1;
+
+    i2c.readReg(deviceID, tempReg, regLength, tempBytes, valueLength);
+
+    byte1 = tempBytes[0];
+    byte1<<=8;
+    input = byte1|tempBytes[1];
```

Comment:
```suggestion
    input = ((uint16_t) tempBytes[0]) << 8 | tempBytes[1];
```
If you do it this way, you can get rid of the variable `byte1`